package NEgoProto;

import "la.proto";

message TOptConfig {
    optional string Method = 1 [default = "LD_MMA"];
    optional uint32 MinimizersNum = 2 [default = 1000];
    optional uint32 MaxEval = 3 [default = 100];
    optional double Tolerance = 4 [default = 1e-10];
}


message TStrategyConfig {
    optional TOptConfig HyperOpt = 7;
    optional TOptConfig AcqOpt = 8;

    optional uint32 InitSamplesNum = 9 [default = 10];
    optional uint32 BatchSize = 10 [default = 1];
    optional uint32 IterationsNum = 11 [default = 100];
    optional uint32 HyperOptFreq = 12 [default = 1];
    optional double HyperLowerBound = 13 [default = -100];
    optional double HyperUpperBound = 14 [default = 100];
};

message TStrategyState {
    optional TStrategyConfig StrategyConfig = 1;
    optional uint32 StartIterationNum = 2;
    optional uint32 EndIterationNum = 3;
    optional uint32 BatchNumber = 4;
    optional TMatrixD InitSamples = 5;
};